Engine/csound_orc_compile.c

```diff
@@ -2344,30 +2555,63 @@ static ARG *create_arg(CSOUND *csound, INSTRTXT *ip, char *s,
 	   s) != NULL) {
     arg->type = ARG_LOCAL;
     setup_arg_for_var_name(csound, arg, ip->varPool, s);
-    if (arg->argPtr == NULL) {
-      csound->Message(csound, Str("Missing local arg: %s\n"), s);
+
+  } else if (s[0] == '#') {
+    const char* t = s + 1; if (*t == '+' || *t == '-') t++;
+    int allDigits = (*t != '\0');
+    for (const char* p = t; *p; ++p) { if (*p < '0' || *p > '9') { allDigits = 0; break; } }
+    if (allDigits) {
+      arg->type = ARG_CONSTANT;
+      arg->argPtr = cs_hash_table_get(csound, csound->engineState.constantsPool, (char*)t);
+      if (arg->argPtr == NULL) {
+        arg->argPtr = find_or_add_constant(csound, engineState->constantsPool, (char*)t,
+                                           cs_strtod((char*)t, NULL));
+      }
+    } else {
+      arg->type = ARG_LOCAL;
+      setup_arg_for_var_name(csound, arg, ip->varPool, s);
+      if (arg->argPtr == NULL) {
+        csoundDie(csound, Str("Missing temporary variable %s in local pool"), s);
+      }
     }
   }
   /* now global vars are searched for */
-  else if(csoundFindVariableWithName(csound, engineState->varPool,
-                                        s) != NULL) {
+  else if(engineState->varPool != NULL && (uintptr_t)engineState->varPool >= 0x1000 &&
+          csoundFindVariableWithName(csound, engineState->varPool, s) != NULL) {
     arg->type = ARG_GLOBAL;
     setup_arg_for_var_name(csound, arg, engineState->varPool, s);
     }
-    else if(csoundFindVariableWithName(csound, csound->engineState.varPool,
-	   s) != NULL) {
+    else if(csound->engineState.varPool != NULL && (uintptr_t)csound->engineState.varPool >= 0x1000 &&
+            csoundFindVariableWithName(csound, csound->engineState.varPool, s) != NULL) {
     arg->type = ARG_GLOBAL;
-    setup_arg_for_var_name(csound, arg, csound->engineState.varPool, s);  
-    
+    setup_arg_for_var_name(csound, arg, csound->engineState.varPool, s);
+
   }
   /* otherwise we have a local arg */
   else {
     arg->type = ARG_LOCAL;
     setup_arg_for_var_name(csound, arg, ip->varPool, s);
     if (arg->argPtr == NULL) {
-      csound->Message(csound, Str("Missing local arg: %s\n"), s);
+      if (s[0] == '#') {
+        csoundDie(csound, Str("Missing temporary variable %s in local pool"), s);
+      } else {
+        /* Soft-fix: if a local variable is referenced before declaration, synthesize it
```

Comment:
Not quite sure of the implication here. Does it mean that a variable use before declaration is now accepted? Can you clarify?